src/main/java/com/example/Library/services/BookService.java

```diff
@@ -74,15 +76,17 @@ public Optional<Book> returnBook(long id) {
                 });
     }
 
-    public Iterable<Book> getAllBooksWithFilterByTitle(String title) {
+    public Page<Book> getAllBooksWithFilterByTitle(Optional<Integer> page, Optional<String> sortBy, String title) {
```

Comment:
Usage of Optionals as input parameters in the service layer is considered to be a bad/contraversial practice, It would be better to handle Optional request arguments in the Controller layer, and then pass concrete values to the service methods.
(See[ related Sonar code smell](https://rules.sonarsource.com/java/RSPEC-3553))